features/exploredash/src/main/java/org/dash/wallet/features/exploredash/ui/PurchaseGiftCardViewModel.kt

```diff
@@ -65,6 +65,11 @@ class PurchaseGiftCardViewModel @Inject constructor(
     var purchaseGiftCardDataMerchant: Merchant? = null
     var purchaseGiftCardDataPaymentValue: Pair<Coin, Fiat>? = null
 
+    var minCardPurchaseCoin: Coin = Coin.ZERO
+    var minCardPurchaseFiat: Fiat = Fiat.valueOf(Constants.USD_CURRENCY, 0)
+    var maxCardPurchaseCoin: Coin = Coin.ZERO
+    var maxCardPurchaseFiat: Fiat = Fiat.valueOf(Constants.USD_CURRENCY, 0)
```

Comment:
We need to keep track of the purchase limits in Fiat and DASH.  The user sees the Fiat values, but internally the `EnterAmountViewModel` needs the limits in DASH (`Coin`).

There is probably a better way to handle this.

Reply:
I seem to recall discussing with @hadia that `EnterAmountFragment` and `EnterAmountViewModel` are not suitable here because of this exact reason. We should've handled this differently - by building this fragment from the ground out of necessary components so that it is similar to `EnterAmountFragment` but with its own logic.

We had somewhat complicated our lives here, but as long as it works it's fine I guess.